plugins/Denizen/scripts/mobs/modifiers/assignment.dsc

```diff
@@ -5,11 +5,17 @@ mob_modifiers_spawned_entity:
         on !player spawns:
             - stop if:<util.random_chance[<server.flag[mob_modifiers.chance]>].not>
             - define build <context.entity.proc[mob_modifiers_get_random_build]>
+            - define build <script[mob_modifier_config].data_key[builds].get[<[build]>]>
             - define valid_prefixes <[build].proc[mob_modifiers_get_valid_prefixes_for_build]>
             - define valid_suffixes <[build].proc[mob_modifiers_get_valid_suffixes_for_build]>
             - if <[valid_prefixes].is_empty> and <[valid_suffixes].is_empty>:
                 - stop
-            - foreach <[valid_prefixes].random[<[build].get[prefixes]>]> as:prefix:
+            - define selected_prefixes <[valid_prefixes].random[<[build].get[prefixes]>]>
+            - define selected_suffixes <[valid_suffixes].random[<[build].get[suffixes]>]>
+            - foreach <[selected_prefixes]> as:prefix:
                 - flag <context.entity> mob_modifiers.<[prefix]>
-            - foreach <[valid_suffixes].random[<[build].get[suffixes]>]> as:suffix:
-                - flag <context.entity> mob_modifiers.<[suffix]>
\ No newline at end of file
+            - foreach <[selected_suffixes]> as:suffix:
+                - flag <context.entity> mob_modifiers.<[suffix]>
+            - define color <&color[<[build].get[color]>]>
+            - define name <[color]><[selected_prefixes].parse_tag[<[parse_value].proc[snake_case_to_title_case]>].space_separated> <&f><context.entity.custom_name.if_null[<context.entity.entity_type.proc[snake_case_to_title_case]>]> <[color]><[selected_suffixes].parse_tag[<[parse_value].proc[snake_case_to_title_case]>].space_separated>
```

Comment:
You could split up this mega-tag

Reply:
Yeah, I'll split it into `formatted_prefix` and then `formatted_suffix` that way the line will be shorter and more readable.

- [ ] Carve mega-tag